src/java.desktop/share/classes/javax/swing/plaf/nimbus/skin.laf

```diff
@@ -25749,12 +25747,7 @@
                      <blendingMode>NORMAL</blendingMode>
                      <locked>false</locked>
                      <visible>true</visible>
-                     <shapes>
-                        <rectangle x1="0.0" x2="100.0" y1="0.0" y2="30.0" rounding="0.0">
-                           <matte red="255" green="255" blue="255" alpha="255" uiDefaultParentName="nimbusLightBackground" hueOffset="0.0" saturationOffset="0.0" brightnessOffset="0.0" alphaOffset="0"/>
-                           <paintPoints x1="0.25" y1="0.0" x2="0.2" y2="0.1"/>
-                        </rectangle>
-                     </shapes>
+		     <shapes/>
```

Comment:
What is the difference between background and shape tags?

Reply:
I have looked at https://docs.oracle.com/javase/tutorial/uiswing/lookandfeel/_nimbusDefaults.html where default color for 
EditorPane.background and TextPane.background is NimbusBlueGrey and not White what we have now, so I think we are having what is being specified in nimbus-default.
SwingSet2 background color for the demos are same before and after the fix.
I am not sure about the difference between background and shape but I have ran full JCK/JTREG tests and all are fine. I also tried passing Nimbus L&F via CI script for JCK/jtreg tests and those are also same before and after fix.

Reply:
In the linke above the 
nimbusBlueGrey 	#a9b0be (169,176,190)
While the "EditorPane.background and TextPane.background" are:
TextPane.background 	#d6d9df (214,217,223)
EditorPane.background 	#d6d9df (214,217,223)

It is still unclear why setting these values in the XML file prevents the user to change it later, also it will be good to understand the purpose of "shape" since it also sets some coordinates values.

Reply:
Yes, I meant the grey background what is being described in EditorPane.background and TextPane.background is what is being shown now, not literally NimbusBlueGrey. Till now we were showing white which is in contrary to this default.

I could not find any purpose of shape by looking at skin.laf as it is not being used in Synth source code, so that is why I ran all JCK/jtreg testsuite and found no regression.  It is also there from initial days.
I leave it for user of Synth/Nimbus L&F  or anyone from open community to highlight the need of "shape"

Reply:
Does it mean that "shapes" tag is not used and we can remove all its usage in the file?

Reply:
I dont know. I am only telling by looking at source code..I dont know what NimbusL&F creator meant it for...

Reply:
I think we should give this fix a chance...sInce this is solving 2 long-standing issue where setBackground() contract is not honoured plus causing no regression to existing tests.
If we do find any testcase with "shape" in future from open community or any other related testcase being failed for this fix, we can always work on it (or at the least roll back this)

Reply:
The last time this code was touched we get a regression https://bugs.openjdk.java.net/browse/JDK-8266510 which was not found by any tests. JCK cannot found them since this is L&F dependent. And regression tests do not have related tests because we had never bugs here, it worked this way from the beginning.

So it will be good to understand why these tags prevent the color change, how the shape tags are used, instead of some unknown changes here and there.

Reply:
JDK-8266510 regression was found by SwingSet2 so it is wrong to say it was not found by any test, maybe it was not verified during fix of JDK-8249674. For this, all existing test is run.